src/components/Layout/index.tsx

```diff
@@ -61,30 +62,64 @@ function Layout(props: LayoutProps) {
     },
   ];
 
+  const agent_routes = [
+    {
+      name: "Tasks",
+      icon: FaClipboardList,
+      matcher: "/agent/tasks",
+    },
+  ]
+
   return (
     <div className="h-screen w-screen">
       {userProfile ? (
         <div className="flex h-full">
           <div className="w-fit shadow-xl rounded-tr-2xl rounded-br-2xl z-40">
             <Sidebar>
               <Sidebar.Items>
-                <Sidebar.ItemGroup>
-                  {huddler_routes.map((route) => {
-                    return (
-                      <div
-                        className={`rounded-lg ${
-                          location.pathname === route.matcher
-                            ? "bg-gray-100"
-                            : ""
-                        }`}
-                      >
-                        <Sidebar.Item href={route.matcher} icon={route.icon}>
-                          {route.name}
-                        </Sidebar.Item>
-                      </div>
-                    );
-                  })}
-                </Sidebar.ItemGroup>
+                {location.pathname.includes("agent") ? (
```

Comment:
Instead of using the pathname, can we pass it down as a prop? like what you have done in the taskView component?

Reply:
Nice catch @kavisha-perera 
But since this is in the layout component it's a bit different from the task view, 
We don't really use the layout component everywhere, it's only called on the app file, 
So there is no way for us to pass down a prop without checking the pathname over there, so either way we have to check the pathname either there or here, 
Top down controlled approach like you suggested is the best way to organise this, but for now can we keep it this way?

Reply:
sorry not Kavisha @YaiyaSam

Reply:
oh yeah you are right. my bad.